lib/ansible/modules/windows/win_zip.ps1

```diff
@@ -0,0 +1,253 @@
+#!powershell
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+#Requires -Version 5.0
```

Comment:
I see no reason why we would need this, having the .NET methods mean that Server 2008 R2 and above will work just fine. Considering we are planning on dropping support for 2008 and 2008 R2 after 2.10 then I'm happy to just remove the `#Requires` statement altogether.

Reply:
Sure, you could do a lot by stringing together some .NET methods using in-line PowerShell, and forgo writing modules all together but is that what you want users to do?  People want to use Ansible for ease of use.  I'm sure we could come up with lots of examples where a module wasn't necessary if you just go dig through the .NET library or COM interfaces and piece it together with multi-line win_shell.  If memory serves, Ansible best practices suggest using a module when one is available and only use shell/win_shell when you have no alternative.  This is intended to be that alternative.  I never quite understood why win_unzip existed but win_zip didn't.

At some point someone got tired of shelling/win_shelling it and decided to write a module to do it properly.  Isn't this why modules are written? To provide methods within the DSL for getting stuff done more easily and quickly and lessen the learning curve for the users?

There actually is basic idempotence with the creates attribute (like the existing win_unzip module).

I'll take a look at the rest of the comments/suggestions.

Thanks.

Reply:
Yep, I was on the fence but having a starting point is better than nothing and a user won't have to know how to do it in .NET. In saying that I think the actual module can be simplified greatly without losing any of the functionality it offers.